### ISSUE TITLE
Don't raise dangerous-default-value when there is no mutation?

### ISSUE THREAD

**Issue by eight04** [PyCQA/pylint#3664]
### Is your feature request related to a problem? Please describe

```py
def foo(a=0, b=0, c=0, bar_options={}):
  return a + b + c + bar(**bar_options)
```

```
test.py:1:0: W0102: Dangerous default value {} as argument (dangerous-default-value)
```

### Describe the solution you'd like

I think it should be safe in this case?

**Comment by eight04** [comment]
pylint 2.5.2
astroid 2.4.1
Python 3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 22:39:24) [MSC v.1916 32 bit (Intel)]

**Comment by PCManticore** [comment]
Thanks for the report. This is a simple check that does not look for mutations by design. The mutations can happen anywhere in that function, including in other function calls, and given pylint is a static analysis tool, it would be close to impossible to figure all the modification sites.
Regardless if it's safe or not, as a safe practice you should avoid mutable defaults.
